### ISSUE TITLE
0.57.4 Discussion

### ISSUE THREAD

**Issue by kelset** [react-native-community/react-native-releases#54]
Conversation on this thread are limited to 0.57.4 release's major issues and backport (cherry-pick) requests from commits that are already on master - so that we can then produce a 0.57.5 release in the near future.

An example of a good such request is a bug fix for a serious issue that has been merged into master but did not make the 0.57.4 cut.

In other words, if you cannot point to a particular commit on master, then your request likely belongs as a new issue in http://github.com/facebook/react-native/issues.

**Comment by kelset** [comment]
Catching up from #48:
* the React sync for 16.6.0 will land in the near future, but I did a couple tests and you should be able to use it without issues already
* The XCode10 related fix commit (https://github.com/facebook/react-native/commit/b44c5ae92eb08125d466cf151cb804dabfbbc690) didn't end up in 0.57.4 because its cherry-picking was super complex. We'll give it another try this for release (in the meantime we workaround has been made more explicit in the changelog to help)
* other commits that have been requested since the list was finalized (so we'll prob attempt to cherry pick this release)
  * https://github.com/facebook/react-native/commit/b002df945b267c6e36aa1041397ffd1c73167037
  * https://github.com/facebook/react-native/commit/09c78fe968e1bb71108c4058e76ebf70178e5a8b

**Comment by mikemorris** [comment]
- Android `textTransform` style support https://github.com/facebook/react-native/commit/22cf5dc5660f19b16de3592ccae4c42cc16ace69

**Comment by mikemorris** [comment]
- Fix incorrect merged asset path with flavor for Android Gralde Plugin 3.2 https://github.com/facebook/react-native/pull/21782

(I'm aware it hasn't been merged yet, but it's a small yet critical fix and been successfully tested by a few users now.)

**Comment by sync** [comment]
There is a flow issue since 0.57.x with View:

```
Cannot instantiate Ref because Ref [1] is not a polymorphic type.

     32│ if (__DEV__) {
     33│   const View = (
     34│     props: Props,
 [1] 35│     forwardedRef: React.Ref<typeof ViewNativeComponent>,
     36│   ) => {
     37│     return (
     38│       <TextAncestor.Consumer>
```

**Comment by sync** [comment]
We are also experience some weird UI glitches on 0.57.x (test 0.57.4) when running test suite with latest detox. Using 0.56.x it doesn't happen.

![glitch-0 57-grey](https://user-images.githubusercontent.com/21725/47532961-8f75fd00-d8fd-11e8-8f81-3131af04fdf2.png)

**Comment by fungilation** [comment]
0.57.x looks like it fixed your UI for you, not the other way around. 😂

**Comment by sync** [comment]
> 0.57.x looks like it fixed your UI for you, not the other way around. 😂

labels were inverted 👍

**Comment by ikesyo** [comment]
I nominate 

- Quote "$NODE_BINARY" in react-native-xcode.sh facebook/react-native@7d4e94e

**Comment by radko93** [comment]
@sync this is not a place to discuss issues, it's only about committs already in `master`.

**Comment by a-tarasyuk** [comment]
Is there any _timeline_ when will be published version **0.57.5**?

**Comment by dryganets** [comment]
It doesn't look like it was backported.
You could use this workaround for now:
```
ReadableNativeArray.setUseNativeAccessor(true);
ReadableNativeMap.setUseNativeAccessor(true);
```

**Comment by kelset** [comment]
I'm still waiting for the React sync commit to land, so until that commit is on master 0.57.5 won't be released.

@mikemorris the PR is currently being imported, hopefully it will land on master soon.

@dryganets I'm not sure I understand what you are referring to 😅

**Comment by ikesyo** [comment]
> the PR is currently being imported, hopefully it will land on master soon.

That was successfully landed. 🎉 So let me nominate the commit:

- Fix incorrect merged asset path with flavor for Android Gradle Plugin 3.2. https://github.com/facebook/react-native/commit/e90319e9fa18661144ad29faf36efba3750edb32

**Comment by radeno** [comment]
Till it will be synced to React 16.6 my two cents to PRs:

- All commits before my comment
- Flow update https://github.com/facebook/react-native/commit/11552a7a7aa6f21c05460a62f0a141bdb7230228
- Update Jest https://github.com/facebook/react-native/commit/1b4fd643256817d29163b37101da612867a225a1
- Metro update. Linearity of commits: 1. https://github.com/facebook/react-native/commit/f867db366aa4f0ead5a20c0d3154ca58be43fc20 2. https://github.com/facebook/react-native/commit/88882951e1607b0af6f1772ef13135e037f9c4e3 3. https://github.com/facebook/react-native/commit/31bb551e75bda155b4821381e5497dc423326e3c
- Update fbjs-scripts https://github.com/facebook/react-native/commit/cdbf719307f41e94a62307ec22463bb562d1c8de

**Comment by Iamsamwhoami** [comment]
Would be nice if we could get https://github.com/facebook/react-native/pull/21179 into 0.57.5, it solves issue https://github.com/facebook/react-native/issues/18997.

**Comment by dryganets** [comment]
@kelset 
I'm talking about commit for this issue:
https://github.com/facebook/react-native/issues/18292

**Comment by brunolemos** [comment]
Suggestion for 0.57.5: https://github.com/facebook/react-native/pull/21283 (https://github.com/facebook/react-native/commit/458d56c0a1ac73c088660830a8bf2db65de7d9a2)

**Comment by kelset** [comment]
@brunolemos that commit is already in 0.57.x since [.2](https://github.com/react-native-community/react-native-releases/blob/master/CHANGELOG.md#removed-features-that-have-been-removed-these-are-breaking-2) -> https://github.com/facebook/react-native/commit/5627616e3f052bcb6a549120335fbdb2cacf8f03

Small update: the sync commit landed during the weekend but it caused some crashes so it was reverted (This is one of the benefits of Facebook running on master 😅).

**Comment by grabbou** [comment]
This commit https://github.com/facebook/react-native/commit/506f92083806e91a90d9a213bcdd05ab3b2ba888 should be cherry-picked to resolve issues caused by a [breaking change](https://github.com/facebook/react-native/commit/506f92083806e91a90d9a213bcdd05ab3b2ba888) that recently shipped. This is to fix multiple crashes that affect RNNavigation by Wix users.

**Comment by ikesyo** [comment]
@grabbou Are you saying about https://github.com/facebook/react-native/commit/b002df945b267c6e36aa1041397ffd1c73167037? That is already nominated in #48 and https://github.com/react-native-community/react-native-releases/issues/54#issuecomment-433143323.

**Comment by fungilation** [comment]
@grabbou, this alongside syncing with React alpha. Can RN generally not cherrypick big changes that's *potentially* breaking on minor releases (0.57.x)? I'm not seeing enough stability on minor releases that's now used for integrating new features. Shouldn't more major changes wait for cherrypicking on the next major release?

My 2c and whether this should be a discussion on a new issue, if there's much to discuss.

**Comment by kelset** [comment]
> Can RN generally not cherrypick big changes that's potentially breaking on minor releases (0.57.x)?

It's about tradeoffs, since 0.58 is going to have some major breaking changes. We'll discuss more about this during the next core meeting happening later this week.

In the meantime, since I haven't got any update on the state of the React sync from the FB team we'll probably test a new 0.57.5 without it because I feel that this release is long overdue (there are already many commits I want to rollout, like the one to restore UIImplementationProvider).

**Comment by radeno** [comment]
About cherry-picking something significant like increasing React version there are two sides (or more).

On one side when is everything OK updating packages should be fine, because if is package based on semantic versioning then updating minor version should be without any disasters. We are using 0.57.4 without any problems. But i agree we are not all.

On the other hand we don't really know how are two software products coupled together in 100%. They could be coupled with non-public API, internal hooks or something else than documented public APIs, like we seen with change `schedule` to `scheduler` in React 16.5.3 and current regression on 16.6.0. Both steps forward breaks another software (React Native). It was not supposed to happen but does.

I am usually happy to touch "newest" technology as quick as possible, anyway stability of (any) product is major priority. Even more if it is just some minor or patch release.

@kelset is RN 0.57.4 compatible with React 16.5.2 ? I don't test it yet. If so, we can add note to changelog, that if user have any issues with 16.6 alpha it can be safe downgraded.
I agree to release 0.57.5 as soon as possible. Some important cherry picks waits. You will see more after discussion with core team.

**Comment by kelset** [comment]
>  is RN 0.57.4 compatible with React 16.5.2 ?
AFAIK no, since 0.57.2 you need to use the 16.6 alpha.

That said, I tried to cherry pick & test this list of commit (order of commit):
- [Incorrect ref API usage fixed.](https://github.com/facebook/react-native/commit/09c78fe968e1bb71108c4058e76ebf70178e5a8b)
- [Fix InputAccessoryView safe area when not attached to a TextInput](https://github.com/facebook/react-native/commit/2191eecf54b5c4bf278dfaf23fec46d44ac9a1f0)
- [Remove previously scheduled NetInfo callbacks if they haven't fired](https://github.com/facebook/react-native/commit/67afaefa78c314b38249a7e2758e0af38c18f34a)
- [Remove unused parameters in ReactInstanceManager methods](https://github.com/facebook/react-native/commit/ef863c06797df80e28c074ae021842a5067cae42)
- [Update size of Root ShadowNode when RootView changes its size](https://github.com/facebook/react-native/commit/8b5e3fc16b1e58441318b6ada629dcff572dd120)
- [Add systrace to UIManagerModule.createConstants](https://github.com/facebook/react-native/commit/cb6196073a7588f27e063a09137994db9779c2de)
- [Fix setupReactContext systrace](https://github.com/facebook/react-native/commit/f38ab24a192a9c2cd0a849fb07d0c7257feb3390)
- [Remove class loads for CoreModulesPackage](https://github.com/facebook/react-native/commit/c31f79fe478b882540d7fd31ee37b53ddbd60a17)
- [Initialize getConstantsForViewManagers eagerly](https://github.com/facebook/react-native/commit/62829a7a6973a452ad69a2ba26b95e662a21655b)
- [Re-introduce UIImplementationProvider](https://github.com/facebook/react-native/commit/b002df945b267c6e36aa1041397ffd1c73167037)
- [RN crash when processing Exception with null message](https://github.com/facebook/react-native/commit/6debfdf6d6172cec2d87fd1e780c3b347d41dc1d)
- [Fix flow in InitializeCore](https://github.com/facebook/react-native/commit/e2210ab752bd978f393dd5c6110626fff92bc6b3)
- [Fix polyfilling of regeneratorRuntime to avoid setting it to undefined in some situations](https://github.com/facebook/react-native/commit/2a7e02edf64c20410b2f95f35e313279545b40db)
- [Don't crash when ReactRootView's id is already set](https://github.com/facebook/react-native/commit/f671f8868f8bf8e7976263e5b2665c2005df7329)
- [Remove hard crash when root view is reused](https://github.com/facebook/react-native/commit/dc74975c5faad1a5f4d3685679394308ca8f6b24)
- [fix android ci](https://github.com/facebook/react-native/commit/99632e16d98c8252d9a61c470558a5acdb1d34fc)
- [Quote "$NODE_BINARY" in react-native-xcode.sh](https://github.com/facebook/react-native/commit/7d4e94edccfc2f642fcbd1d6aa00756f02e3a525)
- [RN: Reduce ForwardRef(View) Noise in Systrace](https://github.com/facebook/react-native/commit/3aea678c38b3f2f208c039a22fc9c6f841826e81)
- [Fix View/Text displayName](https://github.com/facebook/react-native/commit/7a914fcef4ae035221e1f984c104ba20430d6fad)
- [Bump metro@0.49.0](https://github.com/facebook/react-native/commit/31bb551e75bda155b4821381e5497dc423326e3c)
- [Bump fbjs-scripts to ^1.0.0](https://github.com/facebook/react-native/commit/cdbf719307f41e94a62307ec22463bb562d1c8de)
- [Wrap measureLayoutRelativeToContainingList in try-catch to mitigate crash](https://github.com/facebook/react-native/commit/580377201793314ca643250c1bd7cf1c47d49920)
- [Fix incorrect merged asset path with flavor for Android Gradle Plugin 3.2](https://github.com/facebook/react-native/commit/e90319e9fa18661144ad29faf36efba3750edb32)
- [gradle repo priority](https://github.com/facebook/react-native/commit/2a349f87f78a75163bc5cfd5e5233bfe00524676)
- [Fix relayout of inline views](https://github.com/facebook/react-native/commit/798517a267841675956cb52a1c0ae493a913962a)
- [Fix ReactRootView mount/unmount race condition](https://github.com/facebook/react-native/commit/309f85ace280184b5458cc701ccb81716b43b789)
- [Remove undefined value on init cli command](https://github.com/facebook/react-native/commit/58732a88b629b40b2d223a76fac46ecee5ae7295)
- [bump buck to 2018.10.29.01. fixes Android CI](https://github.com/facebook/react-native/commit/b40e23dee5a9a563ca46a0a90d42f998fb2c3f38)
- [Added lock around RN module initialization to fix crash](https://github.com/facebook/react-native/commit/6770b53f8d064a6d14cbd8dd8dcfdc27ac98deb0)
- [Added locking around RN bridge cxx module registry to avoid crash](https://github.com/facebook/react-native/commit/1c3191992a0ce408bb1d6e38934ce4b4118bd93c)
- [Performance improvement for loading cached images on iOS](https://github.com/facebook/react-native/commit/54f7eb34243715a1d4bc821ccbadeec12486d22c)
- [Increase cache and file size limits](https://github.com/facebook/react-native/commit/3a98318c91283a1bba35c0bca93b975d4a550330)

But while testing I experienced an hard crash on Android, and after a lot of investigation I understood that this commit [Remove class loads for CoreModulesPackage](https://github.com/facebook/react-native/commit/c31f79fe478b882540d7fd31ee37b53ddbd60a17) is the one introducing it. I have already reached out to the FB developer to help me understanding what am I missing / where is the fix, and hopefully I'll have something fully working locally.

On top of this then I'll manually apply the [Xcode 10 third party commit](https://github.com/facebook/react-native/commit/b44c5ae92eb08125d466cf151cb804dabfbbc690) - since it's the only way to avoid a billion merge conflicts on the Xcode proj file (@radko93 has been helping me with this).

In any case, this new release won't land before the core meeting on Thursday, but I wanted to be transparent so that maybe someone can help understanding what causes that Android error.

**Comment by fungilation** [comment]
Looks like a great list of mostly bug fixes. Thanks @kelset.

At the FB meeting, any possibility to bring up accelerating a sync with React 16.6.x release? I want to upgrade my RN use but I personally still don't want to with a React alpha dependency.

**Comment by hramos** [comment]
@fungilation it's been worked on. These syncs are not quite straightforward to perform, and this time around we're getting more people from across Facebook's JavaScript tooling teams to jump in to see if we can streamline the process for future releases.

In this particular case, we landed the 16.6 sync last week (https://github.com/facebook/react-native/commit/8b275a8b8d0e4eb7d7e47917c4493570f2cefd84), but were forced to revert it due to some critical bugs we encountered (https://github.com/facebook/react-native/commit/6448f4e2f1ff87600c559208f0f31c8e9fcedd9a). We're working hard to move past these issues and land a 16.6 sync in master again.

**Comment by mmccartney** [comment]
@kelset Would it help if I squash my changes under  
https://github.com/facebook/react-native/pull/21458 into a single commit for you?

I'm sure I can handle the `git` operations but not sure where best to place it for cherry-pick.  I could put it in my personal fork if that helps...

**Comment by kelset** [comment]
Update on my side: I've talked with the FB team about the Android crash and by removing some commits that I have cherry picked I managed to fix the issue I posted above. But then I stumbled again on it (= Android crash during test phase) when trying to upgrade to Metro 0.49 (adding the jest 24 alpha commits don't seem to affect the outcome, but I've learned is actually required for Metro to work properly) 😰

The list of commits I managed to pick without issues is:

- [Incorrect ref API usage fixed.](https://github.com/facebook/react-native/commit/09c78fe968e1bb71108c4058e76ebf70178e5a8b)
- [Fix InputAccessoryView safe area when not attached to a TextInput](https://github.com/facebook/react-native/commit/2191eecf54b5c4bf278dfaf23fec46d44ac9a1f0)
- [Remove previously scheduled NetInfo callbacks if they haven't fired](https://github.com/facebook/react-native/commit/67afaefa78c314b38249a7e2758e0af38c18f34a)
- [Remove unused parameters in ReactInstanceManager methods](https://github.com/facebook/react-native/commit/ef863c06797df80e28c074ae021842a5067cae42)
- [Re-introduce UIImplementationProvider](https://github.com/facebook/react-native/commit/b002df945b267c6e36aa1041397ffd1c73167037)
- [RN crash when processing Exception with null message](https://github.com/facebook/react-native/commit/6debfdf6d6172cec2d87fd1e780c3b347d41dc1d)
- [Fix flow in InitializeCore](https://github.com/facebook/react-native/commit/e2210ab752bd978f393dd5c6110626fff92bc6b3)
- [Fix polyfilling of regeneratorRuntime to avoid setting it to undefined in some situations](https://github.com/facebook/react-native/commit/2a7e02edf64c20410b2f95f35e313279545b40db)
- [fix android ci](https://github.com/facebook/react-native/commit/99632e16d98c8252d9a61c470558a5acdb1d34fc)
- [Quote "$NODE_BINARY" in react-native-xcode.sh](https://github.com/facebook/react-native/commit/7d4e94edccfc2f642fcbd1d6aa00756f02e3a525)
- [RN: Reduce ForwardRef(View) Noise in Systrace](https://github.com/facebook/react-native/commit/3aea678c38b3f2f208c039a22fc9c6f841826e81)
- [Fix View/Text displayName](https://github.com/facebook/react-native/commit/7a914fcef4ae035221e1f984c104ba20430d6fad)

And this is the "backlog" I want to have for the release too:

- [jest: upgrade to 24.0.0-alpha.4](https://github.com/facebook/react-native/commit/66aba092514abd2b278a4fb66c30abffbdd5d5ff) plus [Bump metro@0.49.0](https://github.com/facebook/react-native/commit/31bb551e75bda155b4821381e5497dc423326e3c) (current blocker with the Android crash)
- [Bump fbjs-scripts to ^1.0.0](https://github.com/facebook/react-native/commit/cdbf719307f41e94a62307ec22463bb562d1c8de)
- [Wrap measureLayoutRelativeToContainingList in try-catch to mitigate crash](https://github.com/facebook/react-native/commit/580377201793314ca643250c1bd7cf1c47d49920)
- [Fix incorrect merged asset path with flavor for Android Gradle Plugin 3.2](https://github.com/facebook/react-native/commit/e90319e9fa18661144ad29faf36efba3750edb32)
- [gradle repo priority](https://github.com/facebook/react-native/commit/2a349f87f78a75163bc5cfd5e5233bfe00524676)
- [Fix relayout of inline views](https://github.com/facebook/react-native/commit/798517a267841675956cb52a1c0ae493a913962a)
- [Fix ReactRootView mount/unmount race condition](https://github.com/facebook/react-native/commit/309f85ace280184b5458cc701ccb81716b43b789)
- [Remove undefined value on init cli command](https://github.com/facebook/react-native/commit/58732a88b629b40b2d223a76fac46ecee5ae7295)
- [bump buck to 2018.10.29.01. fixes Android CI](https://github.com/facebook/react-native/commit/b40e23dee5a9a563ca46a0a90d42f998fb2c3f38)
- [Performance improvement for loading cached images on iOS](https://github.com/facebook/react-native/commit/54f7eb34243715a1d4bc821ccbadeec12486d22c)
- [Increase cache and file size limits](https://github.com/facebook/react-native/commit/3a98318c91283a1bba35c0bca93b975d4a550330)

That said, if anyone want to take a stab at understanding what's wrong, you can (if you are on a mac):
* checkout the main repo to `0.57-stable`
* use `git cherry-pick [hash of the commit]` to cherry pick the commits in order
* run `./scripts/test-manual-e2e.sh` (make sure you have the NDK17 and an android device or simulator open before running)

@mmccartney thanks for wanting to help, I think that the best thing would be, once  we've managed to cherry pick the whole list without issues, for you to try and replicate the fix on top of `0.57-stable` once we pushed these commits on top of it - so first need to unlock the issue above sadly. (btw if you are on twitter pls [DM me there](https://twitter.com/Kelset) so that we can coordinate more easily)

**Comment by kelset** [comment]
Hey everyone - quick update: I've attempted once more to cherry pick the commits and test locally, and by removing the ones bumping jest, metro and fbjs-script I was able to have everything ✅ locally. So I proceeded to add on top of that the React sync, so that we can move away from the 16.6 alpha into 16.6.1 (which, from my understanding, should also enable hooks - but pls double check) - I've retested everything locally and still ✅.

I've now pushed [this list of commits](https://github.com/facebook/react-native/commits/0.57-stable) to the remote branch for the CI tests.


A couple of other notes:
* @mmccartney kindly offered to do a PR against `0.57-stable` with the Xcode 10 fix, so once that's up, I'll merge (or replicate or something) then I'll retest locally, and then I'll proceed to bump & release 0.57.5
* I'll write up in more details the core meeting in the discussion repo, but one of the things we agreed on is that once this patch lands we won't change version of React or any other package for 0.57.x (also because 0.58 rc0 should be released soon). Another is that we will be more conservative on accepting cherry picks requests. Anyway I'll link the notes once they are up.

**Comment by fungilation** [comment]
Want to confirm, that React 16.6.1 with sync will ship in RN 0.57.5? Hurray if so.

As for being more conservative on accepting cherry picks, I assume that's for the minor version releases? Wouldn't want to slow down progress in the main releases with rc's.

**Comment by hramos** [comment]
I understand it would be for both. From a maintainer's point of view, it's the same amount of work to land a cherry-pick in 0.57.5 as it would be on 0.58.0-rc.1 once the RC is cut. 

I'd venture to say we should be even more conservative about commits that are cherry picked into the RC over a patch version, since the goal of the RC is to ensure the next release is stable: cherry-picks into a RC should strictly fix regressions introduced by the RC cut itself. Take for a example a change that made it into the RC that depends on a newer version of Metro, but the Metro bump commit missed the RC cut - it would be reasonable to cherry-pick the Metro bump commit. On the other hand, a fix for a long standing issue that landed on master after the RC cut can wait for the next patch release once RC hits stable.

This is something that is still in discussion, and keep in mind that maintainers can be convinced otherwise by simply putting in the work. If you strongly believe a commit can't wait for the release, sending a PR to the stable branch with the commit applied cleanly and all tests passing will go a long way towards making your case. Part of the motivation behind this discussion is the fact people often asks for commits to be cherry-picked without verifying if there are other commits that need to be pulled in as well in order to keep tests green.

**Comment by rajivshah3** [comment]
> CI is being a bit annoying, something wrong with Flow and Metro

There's a few issues happening:

> Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ node_modules/metro/src/HmrServer.js.flow:99:37
>
> Cannot call formatBundlingError with new GraphNotFoundError(...) bound to error
because undefined [1] is incompatible with number [2] in property lineNumber.

> Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ node_modules/metro/src/HmrServer.js.flow:110:37
>
> Cannot call formatBundlingError with new RevisionNotFoundError(...) bound to
error because undefined [1] is incompatible with number [2] in property
lineNumber.

> Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ node_modules/metro/src/HmrServer.js.flow:174:13
>
> Cannot call formatBundlingError with new RevisionNotFoundError(...) bound to
error because undefined [1] is incompatible with number [2] in property
lineNumber.

I'm not sure what these mean, but upgrading `flow-bin` to 0.86 seems to fix them. The downside is that the upgrade reports over 40 flow errors (the current one only reports 4).

> Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ node_modules/metro/src/index.js.flow:130:51
>
> Cannot resolve module metro-visualizer.

I think this was fixed in https://github.com/facebook/metro/pull/278 but later removed in https://github.com/facebook/metro/commit/aba22ad391f7b6377cf7f00bc6ca1827c4193abd (I guess a new version of flow was supposed to fix it?). Nonetheless, upgrading `flow-bin` still doesn't fix that issue. I'll keep looking into this, but I thought to post it here in case anyone else is more familiar with flow

**Comment by mmccartney** [comment]
https://github.com/facebook/react-native/pull/22230 has my work from https://github.com/facebook/react-native/pull/21458 done as a single commit (for easy picking) and against the `0.57-stable` branch.  Have a great weekend everyone!

**Comment by kelset** [comment]
Good Monday everyone - and thanks for keeping the conversation moving:

- I second everything Hector said, and again it's an open discussion (the notes of the meeting will land soon, the PR is already up)
- Luckily I have enough permissions in the main repo to be able to merge @mmccartney's PR (thank you so much again!), I'll test it locally 👏
- Thanks @rajivshah3 for your investigation, I can understand that potentially upgrading Flow fixes the issues but that's probably the least favourable way (it would force all the people on 0.57.4 to upgrade their flow too AFAIK) - I'll check if upgrading Metro is a viable solution, and I'll report back. This because AFAIK `react-native init` will actually pull the latest Metro anyway, and that means needing Jest 24 because of a breaking change on Metro 0.48.3 (sigh). Anyway I'll keep you all updated 🤞

**Comment by ptomasroos** [comment]
I will try internally in our circle ci @kelset but here's something public with detox tests https://travis-ci.com/ptomasroos/react-native-idfa/builds/91078892

**Comment by iainbeeston** [comment]
Works for me locally using Xcode 10.1, without running the workaround fix for 3rd party dependencies.

**Comment by radeno** [comment]
For me also.
Xcode 10.1, React 16.5.1, even with metro 0.49

**Comment by a-tarasyuk** [comment]
Xcode 10.1, React 16.6.1 works out of the box, without any additional workarounds

**Comment by ptomasroos** [comment]
All green in our internal ci + test environment, both for android + ios

**Comment by kelset** [comment]
Thanks everyone for the feedback, I'll proceed with the release later today.

@a-tarasyuk as I mentioned in my comment, Android has issues when RN is pulled from GitHub ;)

**Comment by a-tarasyuk** [comment]
@kelset ok ;).

**Comment by ptomasroos** [comment]
We compile everything locally and that shows no sign of issues on android FYI @kelset
